src/Stripe/Infrastructure/Requestor.cs

```diff
@@ -74,41 +102,32 @@ private static string GetAuthorizationHeaderValueBearer(string apiKey)
             return string.Format("Bearer {0}", apiKey);
         }
 
-        private static string ExecuteWebRequest(WebRequest webRequest)
+        private async static Task<string> ExecuteRequest(HttpRequestMessage requestMessage)
         {
-            try
+            var response = await HttpClient.SendAsync(requestMessage);
+            var responseText = await response.Content.ReadAsStringAsync();
+            if (response.IsSuccessStatusCode)
             {
-                using (var response = webRequest.GetResponse())
-                {
-                    return ReadStream(response.GetResponseStream());
-                }
+                return responseText;
             }
-            catch (WebException webException)
+            else
             {
-                if (webException.Response != null)
-                {
-                    var statusCode = ((HttpWebResponse)webException.Response).StatusCode;
-
-                    var stripeError = new StripeError();
-
-                    if(webRequest.RequestUri.ToString().Contains("oauth"))
-                        stripeError = Mapper<StripeError>.MapFromJson(ReadStream(webException.Response.GetResponseStream()));
-                    else
-                        stripeError = Mapper<StripeError>.MapFromJson(ReadStream(webException.Response.GetResponseStream()), "error");
-
-                    throw new StripeException(statusCode, stripeError, stripeError.Message);
-                }
-
-                throw;
+                throw BuildStripeException(response.StatusCode, requestMessage.RequestUri.AbsoluteUri, responseText);
             }
         }
 
-        private static string ReadStream(Stream stream)
+        internal static StripeException BuildStripeException(HttpStatusCode statusCode, string requestUri, string responseContent)
         {
-            using (var reader = new StreamReader(stream, Encoding.UTF8))
-            {
-                return reader.ReadToEnd();
-            }
+            var stripeError = new StripeError();
+
+            if (requestUri.Contains("oauth"))
+                stripeError = Mapper<StripeError>.MapFromJson(responseContent);
+            else
+                stripeError = Mapper<StripeError>.MapFromJson(responseContent);
+
+            return new StripeException(statusCode, stripeError, stripeError.Message);
         }
+
```

Comment:
The only issue I ran into is that Portable libraries can't access System.Configuration config settings. So I had to change the tests to always work off of the passed-in key (which is the change below). Architecturally, there might be a better way configure the Stripe client so that you can interact with it without having to instantiate separate service instances, but we can talk about that later ;)